Tools/LyTestTools/tests/unit/test_o3de_multi_test_framework.py

```diff
@@ -0,0 +1,1184 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+import unittest
+
+import pytest
+import unittest.mock as mock
+
+import ly_test_tools
+import ly_test_tools.launchers.exceptions
+import ly_test_tools.o3de.multi_test_framework as multi_test_framework
+from ly_test_tools.launchers.platforms.win.launcher import WinLauncher
+
+pytestmark = pytest.mark.SUITE_smoke
+
+
+class TestAbstractTestBase(unittest.TestCase):
+
+    def test_BaseAttributes_Init_CorrectAttributes(self):
+        mock_abstracttestbase = multi_test_framework.AbstractTestBase()
+        assert mock_abstracttestbase.timeout == 180
+        assert mock_abstracttestbase.test_module is None
+        assert mock_abstracttestbase.attach_debugger is False
+        assert mock_abstracttestbase.wait_for_debugger is False
+
+    def test_SharedTest_Init_CorrectAttributes(self):
+        mock_sharedtest = multi_test_framework.SharedTest()
+        assert mock_sharedtest.timeout == 180
+        assert mock_sharedtest.test_module is None
+        assert mock_sharedtest.attach_debugger is False
+        assert mock_sharedtest.wait_for_debugger is False
+        assert mock_sharedtest.is_batchable
+        assert mock_sharedtest.is_parallelizable
+
+    def test_ParallelTest_Init_CorrectAttributes(self):
+        mock_paralleltest = multi_test_framework.ParallelTest()
+        assert mock_paralleltest.timeout == 180
+        assert mock_paralleltest.test_module is None
+        assert mock_paralleltest.attach_debugger is False
+        assert mock_paralleltest.wait_for_debugger is False
+        assert not mock_paralleltest.is_batchable
+        assert mock_paralleltest.is_parallelizable
+
+    def test_BatchedTest_Init_CorrectAttributes(self):
+        mock_batchedtest = multi_test_framework.BatchedTest()
+        assert mock_batchedtest.timeout == 180
+        assert mock_batchedtest.test_module is None
+        assert mock_batchedtest.attach_debugger is False
+        assert mock_batchedtest.wait_for_debugger is False
+        assert mock_batchedtest.is_batchable
+        assert not mock_batchedtest.is_parallelizable
+
+
+class TestResultType(unittest.TestCase):
+
+    log_attribute = 'log_attribute'
```

Comment:
minor: Unclear why this needs class-level storage is necessary. If it's just a constant used by the tests, can hint that by capitalizing it ...though it seems like it would be more simple as a string directly in the two tests that use it.

Reply:
this should have been removed, it was part of an older design pattern

the new patterns use `self._log_attribute` from the current test suite class and pass that into the intended result as part of its parameters (i.e. `test_result = Result.Pass(self._log_attribute, test_spec, output, executable_log_content)`)